### PR TITLE
[tests] Pin avocado test runner to known working version

### PR DESCRIPTION
**Pull request by clebergnu**
Avocado has been developing a new test runner (dubbed nrunner) and the
upcoming release 91.0 will activate the new runner as the default one.

While most of the tests and jobs should work under the new runner,
there are a few caveats, and possibly unknown issues.

To avoid regressions in the test execution and CI status, let's pin
the test runner to the known working version.  A switch to use the new
runner may be done as a separate future change after compatibility is
fully investigated and validated.

Reference: https://avocado-framework.readthedocs.io/en/90.0/releases/90_0.html#important-announcement
Signed-off-by: Cleber Rosa <crosa@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by TurboTurtle** [comment]
Would it be better to specify the version we install via pip? E.G. `pip3 install avocado-framework==90`?

**Comment by clebergnu** [comment]
> Would it be better to specify the version we install via pip? E.G. `pip3 install avocado-framework==90`?

That would be a way to also pin to the current runner, but you'd miss any general new fix, improvement or bugs :).

BTW, I made a mistake on the first commit pushed, but I don't expect this change to introduce new test failures.  I'd say we wait for the CI run to finish, and then make a better informed decision about which approach to take.

**Comment by clebergnu** [comment]
> > Would it be better to specify the version we install via pip? E.G. `pip3 install avocado-framework==90`?
> 
> That would be a way to also pin to the current runner, but you'd miss any general new fix, improvement or bugs :).
> 
> BTW, I made a mistake on the first commit pushed, but I don't expect this change to introduce new test failures. I'd say we wait for the CI run to finish, and then make a better informed decision about which approach to take.

TL;DR: CI is happy, as expected.  IMO, it's safe to go with this AS IS at this point.

We could go an extra mile and perform a check with a snapshot of what will be Avocado 91.0 (but that wasn't done for any other release).  Also, the "runner" implementation will be deprecated and eventually removed in the upcoming releases.  Between now and then, we need to check compatibility and address them on either side.

**Comment by TurboTurtle** [comment]
>  Also, the "runner" implementation will be deprecated and eventually removed in the upcoming releases.

That is my reasoning for applying a version specific tag to the `pip install` step. Is there going to be another LTS  before the current implementation is removed? If so, I'd say let's go with that version. When we first picked up avocado we determined we should settle on an LTS release after we work out the major hiccups in our approach, which I'd say we've done at this point.

If it's going to be a bit before the next LTS, then we can pick up this change in the short term and settle on an LTS when it releases. At that point we can either continue using this change or do more thorough testing around the new "nrunner" implementation.

**Comment by clebergnu** [comment]
> > Also, the "runner" implementation will be deprecated and eventually removed in the upcoming releases.
> 
> That is my reasoning for applying a version specific tag to the `pip install` step. Is there going to be another LTS before the current implementation is removed? If so, I'd say let's go with that version. When we first picked up avocado we determined we should settle on an LTS release after we work out the major hiccups in our approach, which I'd say we've done at this point.
> 

I see your point, and I do not disagree with that approach. My point is that by initially pinning runners, you could still leverage new features and fixes while not introducing compatibility issues.

About another LTS release: yes, I anticipate another LTS release before the removal of the "runner" implementation.

So I guess your options include: 
* pinning a version now and working possible issues before pinning to an LTS version
* not pinning a version now, and facing possible issues earlier (at every new release)

> If it's going to be a bit before the next LTS, then we can pick up this change in the short term and settle on an LTS when it releases. At that point we can either continue using this change or do more thorough testing around the new "nrunner" implementation.

Agreed.  The decision is still open on whether to keep the "runner" implementation is going to be in the next LTS release.  IMO it should and I'd bet it will.

So, feel free to NACK this PR (no hard feelings at all), and replace it with another one pinning the Avocado version.  But, I believe this should be done ASAP as 91.0 has already been released.

**Review by TurboTurtle — APPROVED** [review]
Ack, thanks for the details.

I'm good with the proposed change at this point, we can always add version pinning if needed.
